tests/admin_checks/tests.py

```diff
@@ -43,6 +45,11 @@ def check(self, **kwargs):
 )
 class SystemChecksTestCase(SimpleTestCase):
 
+    def setUp(self):
+        super(SystemChecksTestCase, self).setUp()
+        admin.sites.all_sites.clear()
```

Comment:
This isn't great and invites test state leakage - it's directly touching the internals of the code under test and not resetting it to the way it was after the tests. It would be best if a copy of `all_sites` was made and then in `tearDown` it was reset to the original state, at least

Possibly fixed by #6980 anyway